sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/StoreResult.java

```diff
@@ -130,174 +122,12 @@ public StoreResponse toResponse(RequestChargeTracker requestChargeTracker) {
     }
 
     private static void setRequestCharge(StoreResponse response, CosmosException cosmosException, double totalRequestCharge) {
+        String totalRequestChargeString = Double.toString(totalRequestCharge);
         if (cosmosException != null) {
-            cosmosException.getResponseHeaders().put(HttpConstants.HttpHeaders.REQUEST_CHARGE,
-                    Double.toString(totalRequestCharge));
-        }
-        // Set total charge as final charge for the response.
-        else if (response.getResponseHeaderNames() != null) {
-            for (int i = 0; i < response.getResponseHeaderNames().length; ++i) {
-                if (Strings.areEqualIgnoreCase(
-                        response.getResponseHeaderNames()[i],
-                        HttpConstants.HttpHeaders.REQUEST_CHARGE)) {
-                    response.getResponseHeaderValues()[i] = Double.toString(totalRequestCharge);
-                    break;
-                }
-            }
+            cosmosException.getResponseHeaders().put(HttpConstants.HttpHeaders.REQUEST_CHARGE, totalRequestChargeString);
+        } else {
+            // Set total charge as final charge for the response.
+            response.getResponseHeaders().put(HttpConstants.HttpHeaders.REQUEST_CHARGE, totalRequestChargeString);
```

Comment:
the logic here also seems a little different, also because of ignoreCase check in the original logic.
Do we ever have a mix use of uppercase/lowercase of the header?